crates/sui-graphql-rpc/src/types/system_state_summary.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    big_int::BigInt, gas::GasCostSummary, safe_mode::SafeMode, stake_subsidy::StakeSubsidy,
+    storage_fund::StorageFund, system_parameters::SystemParameters,
+};
+use async_graphql::*;
+use sui_types::sui_system_state::sui_system_state_summary::SuiSystemStateSummary as NativeSystemStateSummary;
+
+#[derive(Clone, Debug)]
+pub(crate) struct SystemStateSummary {
+    pub native: NativeSystemStateSummary,
+}
+
+/// Aspects that affect the running of the system that are managed by the validators either
+/// directly, or through system transactions.
+#[Object]
+impl SystemStateSummary {
+    /// SUI set aside to account for objects stored on-chain, at the start of the epoch.
+    /// This is also used for object rebates.
+    async fn storage_fund(&self) -> Option<StorageFund> {
+        Some(StorageFund {
+            total_object_storage_rebates: Some(BigInt::from(
+                self.native.storage_fund_total_object_storage_rebates,
+            )),
+            non_refundable_balance: Some(BigInt::from(
+                self.native.storage_fund_non_refundable_balance,
+            )),
+        })
+    }
+
+    /// Information about whether this epoch was started in safe mode, which happens if the full epoch
+    /// change logic fails for some reason.
+    async fn safe_mode(&self) -> Option<SafeMode> {
+        Some(SafeMode {
+            enabled: Some(self.native.safe_mode),
+            gas_summary: Some(GasCostSummary {
+                computation_cost: self.native.safe_mode_computation_rewards,
```

Comment:
not blocking this PR, but I find it strange that we use language like computation_cost  when the underlying field is safe_mode_computation_rewards, or is it just me?

Reply:
It's all a matter of perspective, right? One person's reward is another person's cost.  The switch comes because in GraphQL we decided to create a reusable type to represent these running gas totals, and we are reusing it everywhere, whereas we don't have the same abstraction in the underlying representation.  So maybe there is room for that to change here, but I would do it that way around (unify it in the underlying representation) than be faithful to the names used in `sui-core`, in this case.